staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/etcd.go

```diff
@@ -50,5 +88,154 @@ func NewREST(resource schema.GroupResource, listKind schema.GroupVersionKind, st
+	if err != nil {
+		return nil, err
+	}
+	return scaleObject, err
+}
+
+func (r *ScaleREST) Update(ctx genericapirequest.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc) (runtime.Object, bool, error) {
+	cr, err := r.registry.GetCustomResource(ctx, name, &metav1.GetOptions{})
+	if err != nil {
+		return nil, false, err
+	}
+
+	oldScale, err := scaleFromCustomResource(cr, r.specReplicasPath, r.statusReplicasPath, r.labelSelectorPath)
+	if err != nil {
+		return nil, false, err
+	}
+
+	obj, err := objInfo.UpdatedObject(ctx, oldScale)
+	if err != nil {
+		return nil, false, err
+	}
+	if obj == nil {
+		return nil, false, apierrors.NewBadRequest(fmt.Sprintf("nil update passed to Scale"))
+	}
+
+	scale, ok := obj.(*autoscalingv1.Scale)
+	if !ok {
+		return nil, false, apierrors.NewBadRequest(fmt.Sprintf("wrong object passed to Scale update: %v", obj))
+	}
+
+	specReplicasPath := strings.TrimPrefix(r.specReplicasPath, ".") // ignore leading period
+	if err = unstructured.SetNestedField(cr.Object, int64(scale.Spec.Replicas), strings.Split(specReplicasPath, ".")...); !ok {
+		return nil, false, err
+	}
+	cr.SetResourceVersion(scale.ResourceVersion)
+
+	cr, err = r.registry.UpdateCustomResource(ctx, cr, createValidation, updateValidation)
+	if err != nil {
+		return nil, false, err
+	}
+
+	newScale, err := scaleFromCustomResource(cr, r.specReplicasPath, r.statusReplicasPath, r.labelSelectorPath)
+	if err != nil {
+		return nil, false, apierrors.NewBadRequest(err.Error())
+	}
+	return newScale, false, err
+}
+
+// scaleFromCustomResource returns a scale subresource for a customresource.
+func scaleFromCustomResource(cr *unstructured.Unstructured, specReplicasPath, statusReplicasPath, labelSelectorPath string) (*autoscalingv1.Scale, error) {
+	specReplicasPath = strings.TrimPrefix(specReplicasPath, ".") // ignore leading period
+	specReplicas, found, err := unstructured.NestedInt64(cr.UnstructuredContent(), strings.Split(specReplicasPath, ".")...)
+	if err != nil {
+		return nil, err
+	} else if !found {
+		specReplicas = 1
+	}
+	if specReplicas < 0 {
+		return nil, fmt.Errorf("spec replicas value should be a non-negative integer")
+	}
+
+	statusReplicasPath = strings.TrimPrefix(statusReplicasPath, ".") // ignore leading period
+	statusReplicas, found, err := unstructured.NestedInt64(cr.UnstructuredContent(), strings.Split(statusReplicasPath, ".")...)
+	if err != nil {
+		return nil, err
+	} else if !found {
+		statusReplicas = 0
+	}
+	if statusReplicas < 0 {
+		return nil, fmt.Errorf("status replicas value should be a non-negative integer")
+	}
+
+	labelSelectorPath = strings.TrimPrefix(labelSelectorPath, ".") // ignore leading period
+	labelSelector, found, err := unstructured.NestedString(cr.UnstructuredContent(), strings.Split(labelSelectorPath, ".")...)
+	if err != nil {
+		return nil, err
+	} else if !found {
+		labelSelector = ""
```

Comment:
is an empty selector valid?

Reply:
Yes. From https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/

> Valid label values must be 63 characters or less and must be empty or begin and end with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between.

Also, if not specified, the default value of the string will be `"`.

Reply:
In `ScaleStatus` it is valid. The validating (or invalidation) of empty label selectors belongs into the CR validation, not here. I can imagine scalable CRs that can were well live with empty label selectors.

Reply:
It won't work well for scale subresources that need to work with HPA, though, FWIW.  That's not the only use of scale, so I can see an argument for allowing it to be empty, but it's certainly going to be a strange class of issues.

Reply:
Empty is a valid value for an optional fields. What do you suggest instead?